File: flight-status-ui/src/app/models/flight-status-result.ts
import { UnifiedFlightStatus } from './unified-flight-status';

export interface FlightStatusResult {
  flightNumber: string;
  date: string;
  status: UnifiedFlightStatus;
  message: string;
  providerUsed: string | null;
  lastUpdatedUtc: string | null;
  scheduledDepartureUtc: string | null;
  scheduledArrivalUtc: string | null;
  actualDepartureUtc: string | null;
  actualArrivalUtc: string | null;
  terminal: string | null;
  gate: string | null;
  delayReason: string | null;
}

File: FlightStatus.Api/Domain/Models/FlightStatusResult.cs
using FlightStatus.Api.Domain.Enums;

namespace FlightStatus.Api.Domain.Models;

/// <summary>
/// The normalized, provider-agnostic flight status result returned by the API.
/// AeroTrack-only fields (Terminal, Gate, ActualDepartureUtc, ActualArrivalUtc, DelayReason)
/// are null when the winning provider is QuickFlight or when the field is not applicable.
/// </summary>
public record FlightStatusResult(
    string FlightNumber,
    DateOnly Date,
    UnifiedFlightStatus Status,
    string Message,
    string? ProviderUsed,
    DateTime? LastUpdatedUtc,
    DateTime? ScheduledDepartureUtc,
    DateTime? ScheduledArrivalUtc,
    DateTime? ActualDepartureUtc,    // AeroTrack conditional
    DateTime? ActualArrivalUtc,      // AeroTrack conditional
    string? Terminal,                // AeroTrack conditional
    string? Gate,                    // AeroTrack conditional
    string? DelayReason              // Only when Delayed and provider supplies it
);
